x-pack/plugins/ml/common/util/time_formatter.ts

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { formatDate } from '@elastic/eui/lib/services/format';
+import { TIME_FORMAT } from '../constants/time_format';
+
+export const timeFormatter = (value: number) => {
```

Comment:
this is very similar to the functions in https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/public/application/util/date_utils.ts

Also we have quite a few places that have almost identical functions to this, which are local to the file and not shared.
https://github.com/elastic/kibana/blob/addbdf7cb608ef1b1aeda1d31d0341f6ace02c98/x-pack/plugins/ml/public/application/components/model_snapshots/model_snapshots_table.tsx#L249

https://github.com/elastic/kibana/blob/addbdf7cb608ef1b1aeda1d31d0341f6ace02c98/x-pack/plugins/ml/public/application/components/annotations/annotations_table/annotations_table.js#L380

plus a handful of places where we're doing something like this:
https://github.com/elastic/kibana/blob/addbdf7cb608ef1b1aeda1d31d0341f6ace02c98/x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/models_management/expanded_row.tsx#L64

I think it be worth moving `date_utils.ts` to common and merging with this file and then changing all uses of `formatDate` were it's being using with `TIME_FORMAT `
I don't know if all of that work would be too out of scope for this PR, if so, then maybe just the merging of `date_utils.ts` would be enough for now.

Reply:
Refactored in https://github.com/elastic/kibana/pull/79072/commits/35b1c8c3cbbef9ee25de52e5454bcbf4d93a4327

- Moved `date_utils` to common
- Introduced a `timeFormatter` based on `TIME_FORMAR` in `date_utils` and replaced all duplicates